apps/core/templatetags/core_tags.py

```diff
@@ -20,3 +20,41 @@ def navigation(context):
         "current_page": context.get("page"),
         "annotated_list": Page.get_annotated_list_qs(pages),
     }
+
+
+@register.filter
+def next_page(value):
+    page = value
+    if page.get_descendants():
+        return page.get_descendants().live().first()
+    elif page.get_siblings().live().filter(path__gt=page.path):
+        return page.get_siblings().live().filter(path__gt=page.path).first()
+    elif page.get_ancestors().live().last().get_siblings().filter(path__gt=page.path):
+        return (
+            page.get_ancestors()
+            .live()
+            .last()
+            .get_siblings()
+            .filter(path__gt=page.path)
+            .first()
+        )
+    else:
+        return None
+
+
+@register.filter
+def previous_page(value):
+    page = value
+    if page.get_siblings().live().filter(path__lt=page.path):
+        prev_page = (
+            page.get_siblings()
+            .live()
+            .filter(path__lt=page.path)
+            .order_by("-path")
+            .first()
+        )
+        if prev_page.get_descendants().live():
+            return prev_page.get_descendants().live().last()
+        return prev_page
+    else:
+        return page.get_ancestors().live().last()
```

Comment:
> I'm of the opinion that on the general category pages, it makes sense to give folks a button to go to the first descendant within the category, but returning to the last page or the last descendant won't be very helpful for users.

Honestly, I think it's fine. Wagtail's docs also does this, and I've found other docs that have the same behaviour (e.g. [Tailwind](https://tailwindcss.com/docs/), [new React docs](https://beta.reactjs.org/learn), [Vue](https://vuejs.org/guide/)).

Reply:
Yes, I also implemented the logic keeping Wagtail's docs in mind. However I am fine with both the styles. Though in the above suggested change, in `previous_page` there is 
```python
if page.get_descendants():
        return None
```
this can cause confusion when there are multiple levels of nesting inside any particular category like `Tutorial, etc.`